js/app/controllers/calcontroller.js

```diff
@@ -233,9 +233,24 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 							createAndRenderEvent(result.calendar, result.vevent.data, view.start, view.end, $scope.defaulttimezone);
 						}
 					}).catch(function(reason) {
-						if (reason === 'delete') {
-							deleteAndRemoveEvent(vevent, fcEvent);
+						if (reason !== 'delete') {
+							return true;
 						}
+
+						fc.elm.fullCalendar('removeEvents', fcEvent.id);
+						fcEvent.delete().then(function() {
+							return VEventService.delete(vevent).catch(function() {
+								throw new Error();
+							});
+						}).catch(function() {
+							const start = view.start.subtract(1, 'day');
+							const end = view.end.add(1, 'day');
+
+							const eventsToRender = vevent.getFcEvent(start, end, $scope.defaulttimezone);
+							angular.forEach(eventsToRender, function (event) {
+								fc.elm.fullCalendar('renderEvent', event);
+							});
+						});
```

Comment:
Rerendering undeleted events like this causes them to double render when another event is created. I know that is a confusing way to word it so I took a shot video (sorry about the screen tearing...)
![Double render](https://cloud.githubusercontent.com/assets/2264352/19708691/59555db4-9ad5-11e6-879b-4dd31d3140e2.gif)

Fortunately, this is simply a rendering artifact and resolves itself on reload. 

If you rerender the events like this:
`fc.elm.fullCalendar('refetchEventSources', vevent.calendar.fcEventSource);`
it does not appear to cause this problem.